planning_through_contact/simulation/controllers/cylinder_actuated_controller.py

```diff
@@ -0,0 +1,92 @@
+from abc import abstractmethod
+from typing import Dict, Any
+
+from pydrake.all import (
+    DiagramBuilder,
+    MultibodyPlant,
+    InverseDynamicsController,
+    StateInterpolatorWithDiscreteDerivative,
+    System,
+    Diagram,
+)
+
+from planning_through_contact.simulation.planar_pushing.planar_pushing_diagram import (
+    PlanarPushingSimConfig,
+)
+
+from .position_controller_base import PositionControllerBase
+from planning_through_contact.simulation.sim_utils import GetParser
+
+
+class CylinderActuatedController(PositionControllerBase):
+    """Base controller class for an actuated floating cylinder robot."""
+
+    def __init__(
+        self,
+        sim_config: PlanarPushingSimConfig,
+    ):
+        self._sim_config = sim_config
+        self._meshcat = None
+        self._pid_gains = dict(kp=800, ki=100, kd=50)
+        self._num_positions = 2  # Number of dimensions for robot position
+
+    def setup(
```

Comment:
Drake convention for adding a system to a diagram with the DiagramBuilder is `AddToBuilder`. If this is what this function is doing, I suggest we change the name.

Reply:
see above